engine/src/main/java/org/terasology/i18n/StandardTranslationProject.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.i18n;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.terasology.i18n.assets.Translation;
+
+/**
+ * Performs textual translations based on a set of {@link Translation} instances.
+ */
+public class StandardTranslationProject implements TranslationProject {
+
+    private final Map<Locale, Translation> translations = new HashMap<>();
+
+    @Override
+    public void add(Translation trans) {
+        translations.put(trans.getLocale(), trans);
+    }
+
+    @Override
+    public void remove(Translation trans) {
+        translations.remove(trans.getLocale(), trans);
+    }
+
+    @Override
+    public String translate(String key, Locale locale) {
```

Comment:
Suggest this should return Optional<String>, or return the key if translation fails.